src/Parsers/Access/ParserGrantQuery.cpp

```diff
@@ -332,7 +332,7 @@ bool ParserGrantQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
         parseOnCluster(pos, expected, cluster);
 
     std::shared_ptr<ASTRolesOrUsersSet> grantees;
-    if (!parseToGrantees(pos, expected, is_revoke, grantees))
+    if (!parseToGrantees(pos, expected, is_revoke, grantees) && !allow_no_grantees)
         return false;
```

Comment:
Probably it's better to move this logic to UsersConfigAccessStorage next to checks about roles and throw an exception with a user-friendly error message.

Reply:
This doesn't throw an exception if grantees are specified. Instead, this bool option just allows the parser not to fail if no grantees were provided. And yes, I definitely forgot to check if grantees are empty like I did with roles

Reply:
Please add a comment about what does that `allow_no_grantees` mean, because it doesn't seem obvious.